web/ui/react-app/src/pages/LabelsTable.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useState } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { Button, Badge, Table } from 'reactstrap';
+import { TargetLabels } from './Services';
+
+interface LabelProps {
+  value: TargetLabels[];
+  name: string;
+}
+
+const formatLabels = (labels: Record<string, string> | string) => {
+  return Object.entries(labels).map(([key, value]) => {
+    return (
+      <div key={key}>
+        <Badge color="primary" className="mr-1">
+          {`${key}="${value}"`}
+        </Badge>
+      </div>
+    );
+  });
+};
+
+export const LabelsTable: FC<RouteComponentProps & LabelProps> = ({ value, name }) => {
+  const [showMore, setShowMore] = useState(false);
+
+  const toggleShowMore = () => {
+    setShowMore(!showMore);
+  };
+
+  return (
+    <div className="label-component">
+      <div className="label-button-style">
+        <span className="target-head"> {name} </span>
+        <Button size="sm" color="primary" onClick={toggleShowMore}>
```

Comment:
Button width now is based on the text size. Will look better if it has a fixed width.

Reply:
My old comment still stands: we should make this look like the same buttons on the /new/targets page by reusing the markup and styling from there.